### PR TITLE
Adding XML Documentation for Generalized Additive Models

### LINKED ISSUE #1730: Add detailed XMLDocs for Generalized Additive Models
The XML docs for Generalized Additive Models are not very detailed and sometimes incorrect (e.g. in the tree trainer catalog). I would like to see detailed descriptions in the XML docs that link to samples.

### PR DESCRIPTION
**Pull request by rogancarr**
This PR updates the XML Docs for GAMs with more information about the learner and points to the new samples being added to the repository.

This also fixes the name of the binary classification trainer.

Fixes #1730

### CODE REVIEW

File: src/Microsoft.ML.FastTree/GamTrainer.cs [hunk 1]
```diff
@@ -34,7 +34,24 @@ namespace Microsoft.ML.Trainers.FastTree
...
+    /// "Intelligible Models for Classification and Regression." KDD�12, Beijing, China. 2012.
+    /// After training, an intercept is added to represent the average prediction over the training set,
+    /// and the shape functions are normalized to represent the deviation from this average prediction.
+    /// <example>
+    /// <format type="text/markdown">
+    /// <![CDATA[
+    /// [!code-csharp[MF](~/../docs/samples/Microsoft.ML.Samples/Dynamic/GeneralizedAdditiveModels.cs)]
+    /// ]]>
+    /// </format>
+    /// </example>
```
Comment (sfilipi) [inline src/Microsoft.ML.FastTree/GamTrainer.cs]: This needs to be outside of <summary>. Also, maybe split the content inside <summary> to one sentence inside <summary>, and the rest inside <remarks>

File: src/Microsoft.ML.FastTree/GamTrainer.cs [hunk 2]
```diff
@@ -34,7 +34,24 @@ namespace Microsoft.ML.Trainers.FastTree
...
-    /// Generalized Additive Model Learner.
+    /// Generalized Additive Model Trainer. Generalized Additive Models, or GAMs, model the data as an
+    /// independent set of feature responses similar to a linear model. For each feature, the GAM trainer
+    /// learns a nonparametric function, known as a shape function, that outputs a response as a function
+    /// of the feature value. (In contrast, a linear model fits a linear response (e.g. a line) to each feature).
+    /// To score an example, the output of each shape function is added up and the total value is the
+    /// output of the model (hence the name Generalized Additive Models).
+    /// This GAM trainer is implemented using shallow gradient boosted trees (e.g. tree stumps) to learn the
+    /// shape functions, and is based on the method described in Lou, Caruana, and Gehrke.
+    /// "Intelligible Models for Classification and Regression." KDD�12, Beijing, China. 2012.
```
Comment (sfilipi) [inline src/Microsoft.ML.FastTree/GamTrainer.cs]: add link

File: src/Microsoft.ML.FastTree/GamTrainer.cs [hunk 3]
```diff
@@ -34,7 +34,24 @@ namespace Microsoft.ML.Trainers.FastTree
...
+    /// After training, an intercept is added to represent the average prediction over the training set,
+    /// and the shape functions are normalized to represent the deviation from this average prediction.
+    /// <example>
+    /// <format type="text/markdown">
+    /// <![CDATA[
+    /// [!code-csharp[MF](~/../docs/samples/Microsoft.ML.Samples/Dynamic/GeneralizedAdditiveModels.cs)]
```
Comment (sfilipi) [inline src/Microsoft.ML.FastTree/GamTrainer.cs]: please see the other link format. .../docs/samples/doc/samples...

Comment (rogancarr) [reply src/Microsoft.ML.FastTree/GamTrainer.cs]: Resolved.

File: src/Microsoft.ML.FastTree/GamTrainer.cs (hunk 2 above)
Comment (rogancarr) [reply src/Microsoft.ML.FastTree/GamTrainer.cs]: Done!

File: src/Microsoft.ML.FastTree/GamTrainer.cs (hunk 1 above)
Comment (rogancarr) [reply src/Microsoft.ML.FastTree/GamTrainer.cs]: Thanks for showing me the right format. Done!

File: src/Microsoft.ML.FastTree/GamTrainer.cs [hunk 4]
```diff
@@ -34,8 +34,33 @@ namespace Microsoft.ML.Trainers.FastTree
...
+    /// independent set of feature responses similar to a linear model. For each feature, the GAM trainer
+    /// learns a function, known as a shape function, that outputs a response as a function
+    /// of the feature value. (In contrast, a linear model fits a linear response (e.g. a line) to each feature).
+    /// To score an example, the output of each shape function is added up and the total value is the
+    /// output of the model (hence the name Generalized Additive Models).
+    /// </para>
+    /// <para>
+    /// This GAM trainer is implemented using shallow gradient boosted trees (e.g. tree stumps) to learn nonparametric
+    /// shape functions, and is based on the method described in Lou, Caruana, and Gehrke.
+    /// <a href='http://www.cs.cornell.edu/~yinlou/papers/lou-kdd12.pdf'>"Intelligible Models for Classification and Regression."</a> KDD�12, Beijing, China. 2012.
```
Comment (sfilipi) [inline src/Microsoft.ML.FastTree/GamTrainer.cs]: >� [](start = 137, length = 1)

Might want to XML escape the apostrophe. Replace by:

&apos;

File: src/Microsoft.ML.FastTree/GamTrainer.cs [hunk 5]
```diff
@@ -34,8 +34,33 @@ namespace Microsoft.ML.Trainers.FastTree
...
     /// <summary>
-    /// Generalized Additive Model Learner.
+    /// Generalized Additive Model Trainer.
     /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Generalized Additive Models, or GAMs, model the data as an
+    /// independent set of feature responses similar to a linear model. For each feature, the GAM trainer
+    /// learns a function, known as a shape function, that outputs a response as a function
+    /// of the feature value. (In contrast, a linear model fits a linear response (e.g. a line) to each feature).
```
Comment (wschin) [inline src/Microsoft.ML.FastTree/GamTrainer.cs]: There are several ambiguous terms, for example, `shape function`, `response`, and `independent set`. Are all those terms consistent with the paper referenced here? Not sure if users would be happy if there is a simple regression example to explain its prediction rule (not training algorithm).

File: src/Microsoft.ML.FastTree/GamTrainer.cs (hunk 4 above)
Comment (rogancarr) [reply src/Microsoft.ML.FastTree/GamTrainer.cs]: Done.

File: src/Microsoft.ML.FastTree/GamTrainer.cs (hunk 5 above)
Comment (rogancarr) [reply src/Microsoft.ML.FastTree/GamTrainer.cs]: Thanks for the close read! `shape function` and `response` are consistent with the literature. `independent set` was not -- I changed this to `linearly independent`. That was what I was looking for :)

**Pull request merged**